export_logic.go

```diff
@@ -0,0 +1,366 @@
+package libsquash
+
+import (
+	"archive/tar"
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+)
+
+var uuidRegex = regexp.MustCompile("^[a-f0-9]{64}$")
+
+func (e *export) IngestImageMetadata(instream io.Reader) error {
+	var tarReader = tar.NewReader(instream)
+	for {
+		header, err := tarReader.Next()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+
+		if header.Name == "." || header.Name == ".." || header.Name == "./" {
+			continue
+		}
+
+		nameParts := strings.Split(header.Name, string(os.PathSeparator))
+		if len(nameParts) == 0 {
+			continue
+		}
+
+		if len(nameParts) == 1 {
+			if nameParts[0] == "repositories" {
+				if err = json.NewDecoder(tarReader).Decode(&e.Repositories); err != nil {
+					return err
+				}
+			}
+
+			// Export may have multiple branches with the same parent.
+			// We can't handle that currently so abort.
+			for _, v := range e.Repositories {
+				commits := map[string]string{}
+				for tag, commit := range *v {
+					commits[commit] = tag
+				}
+				if len(commits) > 1 {
+					return errorMultipleBranchesSameParent
+				}
+			}
+
+			continue
+		}
+
+		uuidPart := nameParts[0]
+		fileName := nameParts[1]
+		if e.Layers[uuidPart] == nil {
+			e.Layers[uuidPart] = &layer{}
+		}
+
+		switch fileName {
+		case "json":
+			e.Layers[uuidPart].JSONHeader = header
+			if err = json.NewDecoder(tarReader).Decode(&e.Layers[uuidPart].LayerConfig); err != nil {
+				return err
+			}
+		case "layer.tar":
+			layerReader := tar.NewReader(tarReader)
```

Comment:
Can this case be broken into a separate function so the logic isn't a for loop within a case statement within a for loop? It would be a bit easier to read and reduce the complexity handled in `IngestImageMetadata`